src/layer/tile/TileLayer.js

```diff
@@ -140,6 +148,10 @@ export var TileLayer = GridLayer.extend({
 			tile.crossOrigin = this.options.crossOrigin === true ? '' : this.options.crossOrigin;
 		}
 
+		if (this.options.referrerPolicy || this.options.referrerPolicy === '') {
+			tile.referrerPolicy = this.options.referrerPolicy === true ? '' : this.options.referrerPolicy;

```

Comment:
Copy-pasting went a bit too far :-)

The `crossOrigin` behaviour accepts an empty string, since that's one of the possible values for the HTML `crossorigin` attribute is an empty string.

But AFAICS, [the HTML `referrerpolicy` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-referrerpolicy) is different. There's no semantics for an empty string, and I don't think it makes much sense to accept a `true` `Boolean` here. It makes no sense to allow or provide) empty strings as the value of the attribute.

Also, note that `referrerpolicy` is lowercase in the MDN docs.

Reply:
> Also, note that referrerpolicy is lowercase in the MDN docs.

Same as for [crossOrigin](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-crossorigin). The html tag is lowercase but the JavaScript tag is `referrerPolicy `

Reply:
So as @Falke-Design noted, there's an HTML attribute vs. DOM property camel-casing distinction here.  So my JS version here is correct as far as the name itself goes.

And admittedly you caught an issue that I wondered about re. the empty string. I glossed over it because it's sort of a common idiom (*attribute*-wise) for "true" and "present but empty" to be analogous and I figured it was harmless. But since you noticed, I went ahead and looked more deeply into it and — now I still think my code is okay :-)

Or at least, just as correct as the `crossOrigin` I copy-pasted from. For [both](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-referrerpolicy) [properties](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-crossorigin) the spec has the same "IDL attribute must reflect the content attribute" text and in both cases a "missing value default" is defined re. the empty string:

* https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attribute
* https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-empty-string

Basically afaict, an empty string is just as useful [or not] for *both* `crossOrigin` and `referrerPolicy` and basically means "keep the default" and so while I considered using `if (typeof this.options.referrerPolicy === 'string')` in the end I opted to match the current approved code as a "least surprise" thing so the two blocks for the two properties would simply match each other.

Reply:
So the lower/uppercase confusion is on me. We should be looking at https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/referrerPolicy and not at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-referrerpolicy . Got it.

But the empty string...

In `crossOrigin`, the lack of value means that the CORS mode is `no-cors`, and a value of an empty string means the CORS mode is `anonymous`. So, in the `L.TileLayer` implementation, `false` becomes `no-cors`, and `true` becomes `anonymous`.

But in `referrerPolicy`, a lack of value means "use the global default for the document", and an empty string means "use the global default for the document". So, in your implementation, `false` becomes "use the default", and `true` becomes.... "use the default". Setting the option from `false` to `true` does nothing, and I don't think that should be the documented behaviour.

Reply:
> Setting the option from false to true does nothing, and I don't think that should be the documented behaviour.

Reasonable enough, I'll make the code match the (already) documented behavior.

Reply:
I think this is resolved now, it now follows the "If a String is provided…" claim the documentation originally made and I've added a brief code comment acknowledging an intentional difference between the divergent implementation of this new one vs. the older `crossOrigin` lines above right above to avoid any confusion in future maintenance.